tests/foundry/src/IBC.t.sol

```diff
@@ -0,0 +1,192 @@
+                client_id: "mock-client-0",
+                connection_id: "connection-0",
+                prefix: MerklePrefix.Data({key_prefix: bytes("ibc")})
+            })
+        });
+        host.setConnection("connection-0", connection);
+        host.setIBCModule(prev);
+    }
+
+    function setUpChannel() internal {
+        address prev = host.getIBCModule();
+        host.setIBCModule(address(this));
+        string[] memory hops = new string[](1);
+        hops[0] = "connection-0";
+        Channel.Data memory channel = Channel.Data({
+            state: Channel.State.STATE_OPEN,
+            ordering: Channel.Order.ORDER_UNORDERED,
+            counterparty: ChannelCounterparty.Data({
+                port_id: portId,
+                channel_id: "channel-0"
+            }),
+            connection_hops: hops,
+            version: "1"
+        });
+        host.setChannel(portId, "channel-0", channel);
+        host.setNextSequenceSend(portId, "channel-0", 1);
+        host.setNextSequenceRecv(portId, "channel-0", 1);
+        host.setNextSequenceAck(portId, "channel-0", 1);
+        host.setIBCModule(prev);
+
+        testPacketCommitment = makePacketCommitment(getPacket());
+    }
+
+    function setUpMockApp() internal {
+        mockApp = new MockApp();
+        handler.bindPort(portId, address(mockApp));
+
+        address prev = host.getIBCModule();
+        host.setIBCModule(address(this));
+        host.claimCapability(IBCIdentifier.channelCapabilityPath(portId, "channel-0"), address(mockApp));
+        host.claimCapability(IBCIdentifier.channelCapabilityPath(portId, "channel-0"), address(this));
+        host.setIBCModule(prev);
+    }
+
+    /* test cases */
+
+    function testClientRegistration() public view {
+        (address client, bool found) = host.getClientImpl(mockClientType);
+        assert(found);
+        assert(address(mockClient) == client);
+    }
+
+    function testSendPacket() public {
+        Packet.Data memory packet = Packet.Data({
+            sequence: 1,
+            source_port: portId,
+            source_channel: "channel-0",
+            destination_port: portId,
+            destination_channel: "channel-0",
+            data: bytes("{\"amount\": \"100\"}"),
+            timeout_height: Height.Data({revision_number: 0, revision_height: 100}),
+            timeout_timestamp: 0
+        });
+        handler.sendPacket(packet);
+    }
+
+    event MockRecv(bool ok);
+
+    function testRecvPacket() public {
+        Packet.Data memory packet = Packet.Data({
+            sequence: 1,
+            source_port: portId,
+            source_channel: "channel-0",
+            destination_port: portId,
+            destination_channel: "channel-0",
+            data: bytes("{\"amount\": \"100\"}"),
+            timeout_height: Height.Data({revision_number: 0, revision_height: 100}),
+            timeout_timestamp: 0
+        });
+
+        vm.expectEmit(true, false, false, true);
```

Comment:
Is `vm.expectEmit(false, false, false, true)` more suitable to your intension?

Reply:
good catch, I fixed it.